swap/swap.go

```diff
@@ -169,12 +170,27 @@ func New(dbPath string, prvkey *ecdsa.PrivateKey, backendURL string, params *Par
 	auditLog.Info("Using backend network ID", "ID", chainID.Uint64())
 	// create the owner of SWAP
 	owner := createOwner(prvkey)
+	// determine the factory address
+	if (factoryAddress == common.Address{}) {
+		if factoryAddress, err = contract.FactoryAddressForNetwork(chainID.Uint64()); err != nil {
+			return nil, err
+		}
+	}
+	factory, err := contract.FactoryAt(factoryAddress, backend)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
A comment would be great here (but might be included in the comment on line 178

Reply:
done